### PR TITLE
NAS-125561 / 24.04 / REST API audit

### PR DESCRIPTION
**Pull request by themylogin**
`audit_callback` support

### CODE REVIEW

**Comment by themylogin** [comment]
time 60:00

**Comment by themylogin** [comment]
time 60:00

**Comment by themylogin** [comment]
https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1443/

File: tests/api2/test_audit_websocket.py [hunk 1]
```diff
@@ -164,6 +141,7 @@ def test_valid_call():
                     "minor": 1,
                 },
                 "origin": ANY,
+                    "protocol": "WEBSOCKET",
```
Comment (anodos325) [inline tests/api2/test_audit_websocket.py:144]: indentation?

**Comment by themylogin** [comment]
@anodos325 @yocalebo @Qubad786 @sonicaj ping

**Review by anodos325 — APPROVED** [review]

**Pull request merged**